src/AppUtils/GlobalEvents.java

```diff
@@ -67,25 +67,36 @@ public static synchronized GlobalEvents instance() {
 
     //<editor-fold defaultstate="collapsed" desc="Reasons to Close">
     public static final int Standard_Close_Operation = 0;
-    public static final int Big_Red_Button = 1;
+    public static final int Application_Restarting = 1;
     //</editor-fold>
     /**
      * <p>
      * Firing this event lets all GlobalEventListeners know that the application
-     * is closing and that they should finalise before calling exit.</p>
+     * is closing and that they should finalise before calling if specified
+     * exit.</p>
      *
-     * @param reason The reason that the application is closing.
+     * @param reason          The reason that the application is closing.
+     * @param exitApplication True if the exit should be called before this
+     *                        method exits.
      */
-    public synchronized void applicationClosing(final int reason) {
+    public synchronized void applicationClosing(final int reason,
+                                                final boolean exitApplication) {
         final GlobalEventListener[] ls = getGobalEventListeners();
         if (ls != null) {
             for (final GlobalEventListener l : ls) {
                 l.applicationClosing(reason);
             }
         }
```

Comment:
Plenty of information given when the application closes now. Should be helpful in debugging.